### PR TITLE
Add is_planar function.  Solves issue #5109

### PR DESCRIPTION
**Pull request by dtekinoglu**
By the suggestion of @rossbar , I implemented `is_planar()` function which is mentioned in issue [#5109](https://github.com/networkx/networkx/discussions/5109).

To Do for myself:
- Add tests for the new method. Could you please guide me on this?

### CODE REVIEW

**Review by rossbar — COMMENTED** [review]
Very nice @dtekinoglu , I think this looks good - just a couple comments on calling `check_planarity` directly and some docstring ideas.

**Comment by rossbar** [comment]
> Add tests for the new method. Could you please guide me on this?

tl;dr - I'd argue we don't need to add any new tests for this function; see below for details.

Great question! Generally yes, whenever a new function is added it needs unit tests, though I think this case is special. Since `is_planar` is really just a wrapper around `check_planarity`, we don't necessarily to add any new tests as the unit test for `check_planarity` should sufficiently cover this function as well (this is one of the advantages of calling `check_planarity` in `is_planar` instead of re-implementing it!)

Note also that the `Examples` section in the docstring doubles as unit tests thanks to `doctest` - I think those are sufficient to ensure that the API is correct (i.e. single input, single output). Nevertheless, if you do think of interesting test cases you'd like to propose, please don't hesitate to add them!

**Review by rossbar — COMMENTED** [review]
The function itself + the docstring LGTM. Since we're adding a new function, there are a couple other things that should be done to better integrate it with the library
- [x] Add `is_planar` to `__all__` for the planarity module (which you've already done - great!)
- [ ] Add `is_planar` to the `.. autosummary::` [here](https://github.com/networkx/networkx/blob/main/doc/reference/algorithms/planarity.rst) so that it shows up in the [reference guide](https://networkx.org/documentation/latest/reference/algorithms/planarity.html)
- [ ] Add `is_planar` to the See Also sections of at least the `PlanarEmbedding` and `check_planarity` docstrings

Note that some of these changes may end up causing merge conflicts since we've already merged a few PRs updating planarity, so you may need to rebase on/merge with `main` before making them. LMK if you have questions about that!

**Comment by dtekinoglu** [comment]
@rossbar I rebased on main and made the requested changes. There are many commits came to this branch as a result of rebaseing. I hope this is something expected. Thank you!

**Comment by dtekinoglu** [comment]
Well, after some effort, I could finalize the PR (hopefully:)) Please let me know, if anything is wrong.

File: networkx/algorithms/planarity.py [hunk 1]
```diff
@@ -49,6 +81,11 @@ def check_planarity(G, counterexample=False):
     A counterexample is only generated if the corresponding parameter is set,
     because the complexity of the counterexample generation is higher.
 
+    See also
+    --------
+    is_planar:
+        Returns True if and only if `G` is planar.
```
Comment (rossbar) [inline networkx/algorithms/planarity.py:87]: Just a minor wording suggestion that (IMO) better takes into account the context of the `check_planar` docstring. This is *very* subjective so feel free to ignore!

```suggestion
        Check for planarity without creating a `PlanarEmbedding` or counterexample.
```

**Review by rossbar — APPROVED** [review]
LGTM, thanks @dtekinoglu ! I took the liberty of pushing up a few minor formatting/line spacing changes but the content is all the same. I also left a wording suggestion, but it's only a suggestion and not a blocker, so feel free to ignore if you like the original wording better!

**Review by dschult — APPROVED** [review]
This LGTM!
Thanks!
